x-pack/legacy/plugins/siem/public/lib/settings/use_timezone_setting.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore: path dynamic for kibana
+import { timezoneProvider } from 'ui/vis/lib/timezone';
+import { npSetup } from 'ui/new_platform';
+import { useMemo } from 'react';
+
+export const useTimezoneBrowser = () => {
+  const config = npSetup.core.uiSettings;
+  const tzValue = useMemo(() => timezoneProvider(config)(), [timezoneProvider, config]);
```

Comment:
i don't think there's any harm in declaring `timezoneProvider` as a dependency in the use of `useMemo` here , but out of curiosity, can the reference to `timezoneProvider` change after it's imported via the following?:

```
import { timezoneProvider } from 'ui/vis/lib/timezone';
```

Reply:
I am confused here by what you are asking

Reply:
> I am confused here by what you are asking

I understand now, I do not know how to do that